GraphsV2/ProgressBarGraph/index.js

```diff
@@ -0,0 +1,297 @@
+}
+
+const getPercentage = ({
+    barData,
+    width,
+    maxData,
+    usedData
+}) => {
+    return barData[maxData] ? (width * barData[usedData] / barData[maxData]) * 100 / width : barData[usedData];
+}
+
+const getColor = ({
+    barData,
+    width,
+    barColor,
+    colorRange,
+    maxData,
+    usedData
+}) => {
+
+    let setColor = false;
+
+    if (colorRange) {
+        const percentage = getPercentage(barData, width, maxData, usedData);
+        let colorData;
+        colorRange.filter((d) => {
+            if (percentage <= d['upto'] && !setColor) {
+                setColor = true;
+                colorData = d;
+            }
+        });
+
+        return colorData['color'];
+    }
+
+    return barColor;
+}
+
+const Container = styled('div')({
+    fontSize: ({ fontSize } = {}) => fontSize,
+    width: ({ width } = {}) => width,
+    height: ({ height } = {}) => height,
+    padding: '0.6rem',
+    overflowY: 'auto',
+    overflowX: 'hidden',
+    position: 'relative',
+});
+
+const Item = styled('div')({
+    width: ({ width } = {}) => width,
+    height: ({ height } = {}) => height,
+    flex: 1,
+    display: 'flex',
+    flexDirection: 'column',
+});
+
+const UpperText = styled('div')({
+    fontSize: ({ fontSize } = {}) => fontSize,
+    order: 1,
+    marginBottom: '0.2rem'
+});
+
+const BarSection = styled('div')({
+    order: 2,
+    display: 'flex',
+    flexDirection: ({ direction } = {}) => direction,
+});
+
+const InnerBarSection = styled('div')({
+    order: 1,
+    flex: 1,
+    height: ({ height } = {}) => height,
+});
+
+const LowerText = styled('div')({
+    fontSize: ({ fontSize } = {}) => fontSize,
+    order: 2,
+    flex: 'auto',
+    alignSelf: ({ alignSelf } = {}) => alignSelf,
+    marginRight: ({ marginRight } = {}) => marginRight,
+    marginTop: '0.4rem'
+});
+
+const Tooltip = styled('div')({});
+
+const ProgressBarGraph = (props) => {
+    const [customTooltips, setCustomTooltips] = useState({});
+    const [hoveredDatum, setHoveredDataum] = useState(null);
+
+    const {
+        data,
+        width,
+        height,
+        properties
+    } = props;
+
+    useEffect(() => {
+        setCustomTooltips(customTooltip(properties));
+    }, [props.data, props.width, props.height]);
+
+    const setHoveredData = (barData) => {
+        if(barData) {
+            setHoveredDataum(barData);
+        }
+    }
+
+    const {
+        margin,
+        label,
+        display,
+        chartWidthToPixel,
+        minSectionHeight,
+        backgroundColor,
+        maxData,
+        usedData,
+        barColor,
+        colorRange,
+        units,
+        maxDataFormat,
+        defaultRange,
+        fontSize,
+        maxSectionHeight,
+        id
+    } = properties;
+
+    const availableWidth = width - (margin.left + margin.right);
```

Comment:
I am thinking that dimension calculations should be done only once not on every re-render. Let's calculate once on mounting the component and then save and re-use.